### ISSUE TITLE
Have a Date type in addition to the DateTime type

### ISSUE THREAD

**Issue by vespertilian** [dgraph-io/dgraph#2279]
I would like to store a Date in Dgraph without the Time 

Currently when I set a date in Dgraph it gets converted to a DateTime

### Alter Schema
```dgraph
somedate: dateTime @index(day) .
```

### Mutate - Set date without time
```dgraph
{
  set {
     _:mydate <somedate> "2018-03-27" .
  }
}
```

### Query
```dgraph
{
  q(func: has(somedate)) {
    somedate
  }
}
```

### Result - Date with time set to midnight UTC
```json
{
       "somedate": "2018-03-27T00:00:00Z"
}
```

Dgraph automatically adds a time, which make sense as the type is DateTime. 

My issue is that by adding the time there can be unintended consequences when parsing.

### Example:

In the example below I changed the time as I am in Australia, however I have debugged issues previously when living in the uk and dates are suddenly a day earlier as daylight savings rolled around and messed with the offset because a function was passed a DateTime with the time set to T00:00:00Z when the function really just wanted just the date as originally set. 

![datetime](https://user-images.githubusercontent.com/441418/38069161-a8632b00-3360-11e8-9faf-65bacb7c2f95.png)
This code is in Typescript using Luxon for parsing

**!!! foo date with time showing 28th not 27th**

I was thinking of storing the date a just a string or {day, month, year} predicate but I feel it makes more sense to use Dgraphs index for DateTime. 

For now I can loop off the date using the T however it would be great if I could store the date in dgraph without the time as that is my true intent.

**Comment by pawanrawal** [comment]
You could get around the issue you are talking about by using UTC date and time? I am not convinced about the use case for adding a separate date type.

**Comment by manishrjain** [comment]
Yeah, explicitly setting UTC would be a way of solving. If you don't care about the actual time, and only care about the day, then using string would be good too. Because any date has to have the right TZ involved, and if you wish to ignore time zones, then string might be a better representation.

**Comment by vespertilian** [comment]
@manishrjain 

How would the string work with indexing? I like the fact I can index on day, month, year with DateTime. I have tried to make a better case below for having a separate date type, if that fails to convince would a string really be the ideal type for indexing dates (without times)? I was thinking maybe a fixed length int (20180301) might be better, your input would be appreciated?

@pawanrawal  

I have been thinking all weekend about how to make a more convincing case for this, as clearly my initial post did not sell it well, and I still feel it would be worthwhile. Here is my current best shot. 

As @manishrjain says "date has to have TZ involved" and by date here I assume we are referring to DGraph's current "DateTime". TZ is something I want to store separately, from date and time, I feel these Date and DateTime are fundamentally different, I also feel that date is a useful type by itself and that's why they deserve to be separate. Kinda like float and int, both are numbers and both are useful for different applications. 

### My definition of date and time

**DateTime** is an exact moment in time, which can be a different date depending on where you are in the world and what timezone you are in. Good for timestamps. 

A **Date**, is a common period the same across the world, but represents different time periods. 

### Example usage

Say you have stores and you want to release a product maybe a movie (or iPhone) at the same time across the world. You want save the release date 27th March 2018 and the time 6pm separately and then combine them into specific DateTimes, for each theatre or store. 

While my application is not exactly this use case this is why I want this feature, I let people select a date, time and timezone and convert these after the fact into a specific DateTime for events. 

### Other reasons

Other DBs seem to have Date (Postgres, MySQL), if someone moves from Postgres to DGraph would be nice to have data consistency where possible. 

Save space, I assume a date type would be more efficient than DateTime as you would need less bits. 

Prevent bugs, as mentioned in the above post I have been bitten by this before when colleagues use DateTimes, where dates would have been more appropriate. Better engineering overall could have fixed this issue but still storing it in the DB as a date would have also fixed it.

**Comment by jLeadr** [comment]
A birthday is a good use case for a date type without a time zone. If you attach time zone to a birthday, it will show up on different days depending on what time and timezone you are looking at the date.
